components/coding/studentPortal/Sidebar.tsx

```diff
@@ -1,16 +1,18 @@
+import { useQuery } from "@apollo/client/react";
 import { link } from "fs";
 import Link from "next/link";
 import { useRouter } from "next/router";
-import React, { useEffect, useRef } from "react";
+import React, { useEffect, useRef, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
+import { FETCH_USER_GOALS } from "../../../graphql/fetchUserGoals";
 import { useAuth } from "../../../lib/authContext";
 import {
   activePageSelector,
   setActivePage,
   SidebarProps,
 } from "../../../redux/sidebarSlice";
 
-const SidebarItem = ({ name, link, page, icon }) => {
+const SidebarItem = ({ name, link, page, icon, number }) => {
```

Comment:
I don't think this is a good approach because we only want to incorporate a number of "urgent" goals for just the Goals sidebar link, not every other page

let's take a step back and brainstorm how to approach this (i honestly don't know the answer to that right now)... also let's get the CEO's design approval before we make a big UI change 😛 

@vithushan19 - for context, we're trying to display some sort of visual cue if a user has X goals due within the next week.... maybe something like this: 
<img width="49" alt="Screen Shot 2022-10-16 at 8 56 58 PM" src="https://user-images.githubusercontent.com/98791180/196068276-3ab19b14-2b86-46d7-9bb4-59c25f087bb7.png">